hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/PmemVolumeManager.java

```diff
@@ -242,16 +262,48 @@ private void loadVolumes(String[] volumes)
       throw new IOException(
           "At least one valid persistent memory volume is required!");
     }
-    cleanup();
   }
 
-  void cleanup() {
+  void cleanup(File realPmemDir) {
     // Remove all files under the volume.
-    for (String pmemDir: pmemVolumes) {
-      try {
-        FileUtils.cleanDirectory(new File(pmemDir));
-      } catch (IOException e) {
-        LOG.error("Failed to clean up " + pmemDir, e);
+    try {
+      FileUtils.cleanDirectory(realPmemDir);
+    } catch (IOException e) {
+      LOG.error("Failed to clean up " + realPmemDir.getPath(), e);
+    }
+  }
+
+  void cleanup() {
+    for (String pmemVolume : pmemVolumes) {
+      cleanup(new File(pmemVolume));
+    }
+  }
+
+  /**
+   * Restore cache in the configured pmem volumes.
+   */
+  public void restoreCache(String bpid) throws IOException {
+    for (byte n = 0; n < pmemVolumes.size(); n++) {
+      long usableBytes = new File(pmemVolumes.get(n)).getUsableSpace();
+      File cacheDir = new File(pmemVolumes.get(n), bpid);
+      if (cacheDir.exists()) {
+        long maxBytes = usableBytes;
+        Collection<File> fileList = FileUtils.listFiles(new File(cacheDir.getPath()),
+            TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE);
+        for (File f : fileList) {
+          maxBytes += f.length();
+          String[] bid = f.getName().split("-");
+          if (bid.length == 5) {
+            long blockId = Long.parseLong(bid[4]);
+            ExtendedBlockId key = new ExtendedBlockId(blockId, bpid);
+            blockKeyToVolume.put(key, n);
+          } else {
+            throw new IOException("Block ID Mismatch");
+          }
+        }
+        usedBytesCounts.get(n).setMaxBytes(maxBytes);
+        cacheCapacity += (maxBytes - usableBytes);
```

Comment:
It looks a bit confused to reader. You can just calculate the figure of cache space used by these restored cache files. And update the maxBytes and overall cache capacity, i.e., add to the previous value. Please also add comment for your purpose.